bundles/org.eclipse.equinox.p2.jarprocessor/src/org/eclipse/internal/provisional/equinox/p2/jarprocessor/JarProcessor.java

```diff
@@ -375,31 +354,24 @@ public File processJar(File input) throws IOException {
 	private void normalize(File input, File directory) {
 		try {
 			File tempJar = new File(directory, "temp_" + input.getName()); //$NON-NLS-1$
-			JarFile jar = null;
-			JarOutputStream jarOut = null;
-			InputStream jarIn = null;
-			try {
-				jar = new JarFile(input, false);
-				jarOut = new JarOutputStream(new BufferedOutputStream(new FileOutputStream(tempJar)));
+			try (JarFile jar = new JarFile(input, false);
+					JarOutputStream jarOut = new JarOutputStream(
+							new BufferedOutputStream(new FileOutputStream(tempJar)))) {
 				Enumeration<JarEntry> entries = jar.entries();
 				for (JarEntry entry = entries.nextElement(); entry != null; entry = entries.hasMoreElements()
 						? (JarEntry) entries.nextElement()
 						: null) {
 					JarEntry newEntry = new JarEntry(entry.getName());
 					newEntry.setTime(entry.getTime());
-					jarIn = new BufferedInputStream(jar.getInputStream(entry));
-					jarOut.putNextEntry(newEntry);
-					Utils.transferStreams(jarIn, jarOut, false);
-					jarOut.closeEntry();
-					jarIn.close();
+					try (InputStream jarIn = new BufferedInputStream(jar.getInputStream(entry))) {
+						jarOut.putNextEntry(newEntry);
+						Utils.transferStreams(jarIn, jarOut, false);
+						jarOut.closeEntry();
```

Comment:
```suggestion
						jarIn.transferTo(jarOut);
```
Explicit closing an entry is not necessary since `putNextEntry()` will close any previously opened entry anyways.